tests/random_attack.py

```diff
@@ -262,6 +267,26 @@ def add_range_column(self, frame):
                                 % (name, rangeobj))
         frame.add_range_column(name, rangeobj)
 
+    def set_key_columns(self, frame):
+        if frame.ncols == 0:
+            return
+        nkeys = random.randint(1, frame.ncols)
+        keys = random.sample(range(0, frame.ncols), nkeys)
+        names = [frame.names[i] for i in keys]
+
+        print("[13] Setting %s: %r"
+              % (plural(nkeys, "key column"), keys))
+
+        if python_output:
+            python_output.write("\ntry:\n"
+                                "   DT.key = %r\n"
+                                "except ValueError as e:\n"
+                                "   assert str(e) == 'Cannot set a key: the "
+                                "values are not unique'\n"
+                                % names)
```

Comment:
Randomly chosen columns may or may not be valid for setting a key. However, at this point it is definitely knowable which case it is. Thus, the python output should be either
```
DT.key = %r
```
or 
```
with check_raises(ValueError('Cannot set a key: the values are not unique')):
    DT.key = %r
``` 
(this context manager can be added trivially).

Reply:
Sure, we can improve this part easily.